gcp/table_gcp_storage_bucket_object.go

```diff
@@ -0,0 +1,349 @@
+package gcp
+
+import (
+	"context"
+	"strings"
+
+	"github.com/turbot/go-kit/helpers"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/storage/v1"
+)
+
+func tableGcpBucketStorageObject(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "gcp_storage_bucket_object",
+		Description: "GCP Storage Object",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"bucket", "name"}),
+			Hydrate:    getBucketStorageObject,
+		},
+		List: &plugin.ListConfig{
+			KeyColumns:    plugin.OptionalColumns([]string{"bucket"}),
+			ParentHydrate: listGcpStorageBuckets,
+			Hydrate:       listBucketStorageObjects,
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the object.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The ID of the object, including the bucket name, object name, and generation number.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "bucket",
+				Description: "The name of the bucket containing this object.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "cache_control",
+				Description: "Cache-Control directive for the object data.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "component_count",
+				Description: "Number of underlying components that make up this object.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "content_disposition",
```

Comment:
Is there any way to get the content of the object itself (or is it in a column already and I missed it)?

Reply:
No API is available to get the content of the object.